### PR TITLE
Implements a large portion of the recompiler for AArch64

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This brings the AArch64 recompiler nearly up to ARMv7 levels of support.
Overall this implements 71 new instructions to the AArch64 recompiler.
With this we get roughly the same performance or slightly better than the ARMv7 recompiler.
Once paired loadstores are implemented then this should be a fair bit quicker.
This requires PR #1849 to be merged prior.

### CODE REVIEW

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStoreFloating.cpp [hunk 1]
```diff
@@ -0,0 +1,394 @@
+// Copyright 2014 Dolphin Emulator Project
+// Licensed under GPLv2
```
Comment (turbedi) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStoreFloating.cpp]: or any later version

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_Paired.cpp [hunk 2]
```diff
@@ -0,0 +1,495 @@
+// Copyright 2014 Dolphin Emulator Project
+// Licensed under GPLv2
```
Comment (turbedi) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_Paired.cpp]: or any later version

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp [hunk 3]
```diff
@@ -0,0 +1,376 @@
+// Copyright 2014 Dolphin Emulator Project
+// Licensed under GPLv2
```
Comment (turbedi) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp]: or any later version

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStoreFloating.cpp (hunk 1 above)
Comment (Sonicadvance1) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStoreFloating.cpp]: We aren't relicensed quite yet.

File: Source/Core/Common/Arm64Emitter.h [hunk 4]
```diff
@@ -565,6 +579,96 @@ class ARM64XEmitter
...
+	void UZP2(u8 size, ARM64Reg Rd, ARM64Reg Rn, ARM64Reg Rm);
+	void TRN2(u8 size, ARM64Reg Rd, ARM64Reg Rn, ARM64Reg Rm);
+	void ZIP2(u8 size, ARM64Reg Rd, ARM64Reg Rn, ARM64Reg Rm);
+
+	// ABI related
+	void ABI_PushRegisters(BitSet32 registers);
+	void ABI_PopRegisters(BitSet32 registers, BitSet32 ignore_mask = BitSet32(0));
+
+private:
+	ARM64XEmitter *_emit;
```
Comment (rohit-n) [inline Source/Core/Common/Arm64Emitter.h]: Asterisk against type.

**Pull request merged**